lib/server/connect.js

```diff
@@ -10,17 +20,29 @@ SubHandlers = {
 var uniqueId = 0;
 var serverId = Random.id();
 
-Meteor.publish('kadira.debug.init', function(browserId, clientId) {
+Meteor.publish('kadira.debug.init', function(browserId, clientId, accessToken) {
   check(browserId, String);
   check(clientId, String);
+  check(accessToken, Match.Any);
 
   var kadiraInfo = Kadira._getInfo();
   if(kadiraInfo) {
     var sessionId = kadiraInfo.session;
 
+    if(AppConfig.env === 'production') {
+      if(!AppConfig.accessToken.has(accessToken)) {
+        // If the subcription request with an invalid accessToken
+        // return without subscribe
+        return;
```

Comment:
You should not return a publication like this. That doesn't stop the publication. Do these:

* Throw a Meteor.Error() (I like that approach)
* Do `return this.stop()`;
* Do `return []` (I don't like this)